src/EntityComponentManager.cc

```diff
@@ -963,6 +963,27 @@ std::unordered_set<ComponentTypeId>
   return periodicComponents;
 }
 
+/////////////////////////////////////////////////
+void EntityComponentManager::EachPeriodicChange(const std::function<
+                  void(const Entity &_entity,
+                       components::BaseComponent *_type)> _f) const
+{
+  for (const auto &[componentType, entities] : this->dataPtr->periodicChangedComponents)
+  {
+    
+    for (const auto entity: entities) {
+
+      const components::BaseComponent* implementation =  
+        this->ComponentImplementation(entity, componentType);
+      
+      if(!implementation)
+        continue;
+      
+      _f(entity, const_cast<components::BaseComponent*>(implementation));
```

Comment:
Do we need the `const_cast`?  Why not change callback the signature to `const components::BaseComponent*`?

Reply:
Thanks for pointing this out I've found a cleaner way to do this in [467048c](https://github.com/gazebosim/gz-sim/pull/2010/commits/467048c5a309ae6e1fbf3e3ab27e51d225ca8145)